src/templates/links.rs

```diff
@@ -0,0 +1,116 @@
+use perseus::prelude::*;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use sycamore::prelude::*;
+
+#[derive(Deserialize)]
+pub struct QueryResult {
+    pub time: Value,
+    pub status: String,
+    pub result: Value,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub struct Link {
+    site: String,
+    url: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct User {
+    pub id: String,
+    pub name: String,
+    pub links: Vec<Link>,
+}
+
+#[perseus::template]
+#[sycamore::component]
+pub fn links_page<G: Html>(cx: Scope, user: User) -> View<G> {
+    let full_name = user.name;
+    let content = create_signal(cx, user.links);
+    view! {
+        cx,
+        h2 { (full_name) }
+        ul{
+            Indexed(iterable=content, view = |cx, link| view!{
+                cx,
+                li{a(href=link.url, rel="external"){( link.site )}}
+            })
+        }
+    }
+}
+
+pub fn get_template<G: Html>() -> Template<G> {
+    Template::new("links")
+        .build_state_fn(get_build_state)
+        .build_paths_fn(get_build_paths)
+        .incremental_generation()
+        .template(links_page)
+}
+
+#[perseus::build_paths]
+pub async fn get_build_paths(path: String, _locale: String) -> RenderFnResult<Vec<String>> {
+    const DB_URL: &str = "http://localhost:8000";
+
+    let mut headers = reqwest::header::HeaderMap::new();
+    headers.insert("Accept", "application/json".parse().unwrap());
+    headers.insert("NS", "weird".parse().unwrap());
+    headers.insert("DB", "weird".parse().unwrap());
+
+    // execute request
+    let client = reqwest::Client::new();
+    let response = client
+        .get(format!("{DB_URL}/key/user"))
+        .basic_auth("root", Some("root"))
+        .headers(headers)
+        .send()
+        .await?;
```

Comment:
When you use `?` you accept that your code will panic when approached with an error. In the future let's replace this with proper error handling